site/app/controllers/admin/PlagiarismController.php

```diff
@@ -667,13 +756,33 @@ public function getRunLog($gradeable_id) {
 
 
     /**
+     * @param string $gradeable_id
+     * @param string $config_id
+     * @param string $user_id_1
+     * @param string $version_user_1
+     * @param string|null $user_id_2
+     * @param string|null $version_user_2
      * @Route("/courses/{_semester}/{_course}/plagiarism/gradeable/{gradeable_id}/concat")
      */
-    public function ajaxGetSubmissionConcatenated($gradeable_id, $user_id_1, $version_user_1, $user_id_2 = null, $version_user_2 = null) {
-        $course_path = $this->core->getConfig()->getCoursePath();
+    public function ajaxGetSubmissionConcatenated(string $gradeable_id, string $config_id, string $user_id_1, string $version_user_1, string $user_id_2 = null, string $version_user_2 = null) {
```

Comment:
This function should probably turn into a JsonResponse rather than just echoing a json_encode.